src/operators_lazytensor.jl

```diff
@@ -96,22 +96,48 @@ isequal(x::LazyTensor, y::LazyTensor) = samebases(x,y) && isequal(x.indices, y.i
 # Arithmetic operations
 -(a::LazyTensor) = LazyTensor(a, -a.factor)
 
-function +(a::LazyTensor{B1,B2}, b::LazyTensor{B1,B2}) where {B1,B2}
-    if length(a.indices) == 1 && a.indices == b.indices
-        op = a.operators[1] * a.factor + b.operators[1] * b.factor
-        return LazyTensor(a.basis_l, a.basis_r, a.indices, (op,))
-    end
-    throw(ArgumentError("Addition of LazyTensor operators is only defined in case both operators act nontrivially on the same, single tensor factor."))
+function +(a::LazyTensor{B1,B2},b::LazyTensor{B1,B2}) where {B1,B2}
```

Comment:
More a generic comment and not strictly necessary in this PR as it works the same: seeing as these methods are quite similar to the `+` methods above, we could (and should) probably create a common abstract type, e.g. `abstract type LazyOperator{BL,BR} <: AbstractOperator{BL,BR} end` and make `LazySum, LazyProduct, LazyTensor <: LazyOperator`. Then we could just define e.g.
```julia
+(a::Operator, b::LazyOperator) = LazySum(a, b)
```
instead of duplicating methods.

Not sure if this will "just work", though, so maybe leave it for another time.

Reply:
I completely agree and have thought about defining that before. I can give it a try and see if it "just works".

Reply:
Could alternatively define `LazyOperator = Union{LazySum,LazyProduct,LazyTensor}`.

Reply:
I ended up choosing LazyOperator as an abstract type. We can change this to LazyOperator = Union{LazySum,LazyProduct,LazyTensor} of course if we want. For now see the following lines where the addition between LazyOperators is defined:

https://github.com/mabuni1998/QuantumOpticsBase.jl/blob/64da84f45beb5fc813b4cc5e4afb5ed7a6bf0fc8/src/operators_lazysum.jl#L82-L104